src/main/java/net/minestom/server/network/packet/server/play/TrackedWaypointPacket.java

```diff
@@ -49,10 +49,33 @@ public record Icon(
         public static final Key DEFAULT_STYLE = Key.key("default");
         public static final Icon DEFAULT = new Icon(DEFAULT_STYLE, null);
 
-        public static final NetworkBuffer.Type<Icon> NETWORK_TYPE = NetworkBufferTemplate.template(
-                NetworkBuffer.KEY, Icon::style,
-                Color.NETWORK_TYPE.optional(), Icon::color,
-                Icon::new);
+        public static final NetworkBuffer.Type<Icon> NETWORK_TYPE = new NetworkBuffer.Type<>() {
+            @Override
+            public void write(@NotNull NetworkBuffer buffer, @NotNull Icon value) {
+                buffer.write(NetworkBuffer.KEY, value.style());
+                final RGBLike color = value.color();
+                buffer.write(NetworkBuffer.BOOLEAN, color != null);
+                if (color != null) {
```

Comment:
this should be defined in Color as another option for color serialization (and use optional here rather than directly in the network type)